UsbIpServer/RegistryWatcher.cs

```diff
@@ -0,0 +1,61 @@
+﻿// SPDX-FileCopyrightText: Microsoft Corporation
+//
+// SPDX-License-Identifier: GPL-2.0-only
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management;
+using System.Threading;
+
+namespace UsbIpServer
+{
+    sealed public class RegistryWatcher : IDisposable
+    {
+        private readonly ManagementEventWatcher? watcher;
```

Comment:
I can live with all the `private` fields/methods, but just like `internal` for classes it is redundant.